packages/transition/src/collapse.tsx

```diff
@@ -139,11 +145,13 @@ export const Collapse = React.forwardRef<HTMLDivElement, CollapseProps>(
       custom,
     }
 
+    const MotionComponent = props.as ?? motion.div
```

Comment:
Perhaps it might be better to add a check if `props.as` is a motion element as well. Maybe throw if it's not a motion element? I'll leave that to you.